### PR TITLE
Fixing ThemeMode.System syncing issues

### PR DESCRIPTION
**Pull request by dipeshmsft**
Fixes # 

Main PR 

###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9558)

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs [hunk 1]
```diff
@@ -204,10 +162,34 @@ internal static bool IsValidThemeMode(ThemeMode themeMode)
...
         bool useLightColors = GetUseLightColors(themeMode);
-        return GetFluentThemeResourceUri(useLightColors);
+        
+        if (themeMode == ThemeMode.System)
+        {
+            rd = new ResourceDictionary() { Source = new Uri(FluentThemeResourceDictionaryUri + "Fluent.xaml", UriKind.Absolute) };
+
+            var colorFileName = useLightColors ? "Light.xaml" : "Dark.xaml";
+            Uri dictionaryUri = new Uri(FluentColorDictionaryUri + colorFileName, UriKind.Absolute);
+            rd.MergedDictionaries.Insert(0, new ResourceDictionary() { Source = dictionaryUri });
```
Comment (harshit7962) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs:184]: At this point we have both system and light/dark dictionary

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Application.cs [hunk 2]
```diff
@@ -932,14 +932,14 @@ public ResourceDictionary Resources
...
                 }
 
-                if(ThemeManager.DeferredAppThemeLoading && !_resourcesInitialized)
+                if(_reloadFluentDictionary && !_resourcesInitialized)
                 {
                     if(value != null)
                     {
-                        var uri = ThemeManager.GetThemeResource(ThemeMode);
-                        value.MergedDictionaries.Insert(0, new ResourceDictionary() { Source = uri });
+                        value.MergedDictionaries.Insert(0, ThemeManager.GetThemeDictionary(ThemeMode));
```
Comment (harshit7962) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Application.cs:939]: Null check needed? `GetThemeDictionary` can return null.

Comment (harshit7962) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Application.cs:939]: Null check needed? `GetThemeDictionary` can return null.

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Application.cs [hunk 3]
```diff
@@ -1739,10 +1749,8 @@ internal void InvalidateResourceReferences(ResourcesChangeInfo info)
             //      i.e. it is not a IsIndividualResourceAddOperation
             //  - the event is not raised due to the change in Application.ThemeMode
             //      i.e. SkipAppThemeModeSyncing is set to true
-            //  - if application's ThemeMode and Resources sync is enabled.
-            //      i.e. IsAppThemeModeSyncEnabled is set to true
-            if (!ThemeManager.SkipAppThemeModeSyncing 
-                    && ThemeManager.IsAppThemeModeSyncEnabled)
+            if (!info.IsIndividualResourceChange
```
Comment (harshit7962) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Application.cs:1752]: As discussed earlier, should we possibly remove this optimization?

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourcesChangeInfo.cs [hunk 4]
```diff
@@ -144,6 +144,16 @@ internal static ResourcesChangeInfo CatastrophicDictionaryChangeInfo
...
+        internal IEnumerable<ResourceDictionary> OldDictionaries
+        {
+            get { return _oldDictionaries; }
+        }
+
+        internal IEnumerable<ResourceDictionary> NewDictionaries
+        {
+            get { return _newDictionaries; }
+        }
+
```
Comment (harshit7962) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourcesChangeInfo.cs:156]: Are these needed in this PR? I don't see them being utilized anywhere.

**Review by harshit7962 — APPROVED** [review]
Tested Multiple scenarios including initialization, theme mode change and theme resource update. Overall changes LGTM.

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourcesChangeInfo.cs (hunk 4 above)
Comment (dipeshmsft) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourcesChangeInfo.cs]: Removed these

**Pull request merged**